account_partial_invoicing/models/sale_order.py

```diff
@@ -0,0 +1,109 @@
+                        continue
+                    if group_key not in invoices:
+                        inv_data = order._prepare_invoice()
+                        invoice = inv_obj.create(inv_data)
+                        references[invoice] = order
+                        invoices[group_key] = invoice
+                    elif group_key in invoices:
+                        vals = {}
+                        if order.name not in invoices[group_key].origin.split(
+                                ', '
+                        ):
+                            vals['origin'] = (
+                                invoices[
+                                    group_key
+                                ].origin + ', ' + order.name
+                            )
+                        if (
+                            order.client_order_ref and
+                            order.client_order_ref not in
+                            invoices[group_key].name.split(', ') and
+                            order.client_order_ref != invoices[group_key].name
+                        ):
+                            vals['name'] = (
+                                invoices[
+                                    group_key
+                                ].name + ', ' + order.client_order_ref
+                            )
+                        invoices[group_key].write(vals)
+                    if line.qty_to_invoice > 0:
+                        line.invoice_line_create(
+                            invoices[group_key].id, line.qty_to_invoice
+                        )
+                    elif line.qty_to_invoice < 0 and final:
+                        line.invoice_line_create(
+                            invoices[group_key].id, line.qty_to_invoice
+                        )
+
+            if references.get(invoices.get(group_key)):
+                if order not in references[invoices[group_key]]:
+                    references[invoice] = references[invoice] | order
+
+        if not invoices:
+            raise UserError(_('There is no invoicable line.'))
+
+        for invoice in invoices.values():
+            if not invoice.invoice_line_ids:
+                raise UserError(_('There is no invoicable line.'))
+            # If invoice is negative, do a refund invoice instead
+            if invoice.amount_untaxed < 0:
+                invoice.type = 'out_refund'
+                for line in invoice.invoice_line_ids:
+                    line.quantity = -line.quantity
+            # Use additional field helper function (for account extensions)
+            for line in invoice.invoice_line_ids:
+                line._set_additional_fields(invoice)
+            # Necessary to force computation of taxes.
+            # In account_invoice, they are triggered
+            # by onchanges, which are not triggered when doing a create.
+            invoice.compute_taxes()
+            invoice.message_post_with_view('mail.message_origin_link',
```

Comment:
@rimbendhaou Not a blocking point but, it would be more readable if you indent the code like this:

```python
invoice.message_post_with_view(
    'mail.message_origin_link',
    values={
        'self': invoice,
        'origin': references[invoice]
    },
    subtype_id=self.env.ref('mail.mt_note').id
)
```